app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx

```diff
@@ -122,12 +123,10 @@ export class DeployStatus extends React.Component<IExecutionDetailsSectionProps,
   private saveManifestSubscription(subscription: IManifestSubscription) {
     const idx = this.findSubscriptionIndex(subscription);
     const subscriptions = [...this.state.subscriptions];
-    if (idx === -1) {
-      subscriptions.push(subscription);
```

Comment:
This line was the crux of the problem.  Any received manifest updates were added to the rendered list even if they didn't have anything to do with the currently selected stage.